examples/qfiae/vqregressor_linear_ansatz.py

```diff
@@ -0,0 +1,372 @@
+
+    def predict_sample(self, features=None):
+        """
+        This function calculates the predictions related to a specific sample
+        Args:
+            features: np.matrix containing the N states; each state must be prepared (with
+                      the opportune self.prepare_states function as an array with dim 2**nqubits)
+        Returns: np.array of the predictions
+        """
+        if features is None:
+            features = self.features
+
+        predictions = np.zeros(len(features))
+        for i in range(len(features)):
+            predictions[i] = self.one_prediction(features[i])
+
+        return predictions
+
+    def one_target_loss(self, this_feature, this_label):
+        """
+        Evaluation of the loss function for a single feature knowing its label
+        Args:
+            this_feature: the feature in form of an 2**nqubits-dim np.array
+            this_label: the associated label
+        Returns: one target loss function's value
+        """
+        this_prediction = self.one_prediction(this_feature)
+        cf = (this_prediction - this_label) ** 2
+        return cf
+
+    def loss(self, params, features=None, labels=None):
+        """
+        Evaluation of the total loss function
+        Args:
+            params: np.array of the params which define the circuit
+            features: np.matrix containig the n_sample-long vector of states
+            labels: np.array of the labels related to features
+        Returns: loss function evaluated by summing contributes of each data
+        """
+        if params is None:
+            params = self.params
+
+        if features is None:
+            features = self.features
+
+        if labels is None:
+            labels = self.labels
+
+        self.set_parameters(params)
+        cf = 0
+        for feat, lab in zip(features, labels):
+            cf = cf + self.one_target_loss(feat, lab)
+        cf = cf / len(labels)
+        return cf
+
+    def dloss(self, features, labels):
+        """
+        This function calculates the loss function's gradients with respect to self.params
+        Args:
+            features: np.matrix containig the n_sample-long vector of states
+            labels: np.array of the labels related to features
+        Returns: np.array of length self.nparams containing the loss function's gradients
+        """
+        loss_gradients = np.zeros(self.nparams)
+        loss = 0
+
+        for feat, label in zip(features, labels):
+            prediction_evaluation = self.one_prediction(feat)
+            loss += (label - prediction_evaluation) ** 2
+            obs_gradients = self.parameter_shift(feat)
+            for i in range(self.nparams):
+                loss_gradients[i] += (2 * prediction_evaluation * obs_gradients[i]) - (
+                    2 * label * obs_gradients[i]
+                )
+
+        return loss_gradients, (loss / len(features))
+
+    def shift_a_parameter(self, i, this_feature):
```

Comment:
As far as I understood, you are uploading your data into the first rotational gate of the circuit. Setting the entire rotational angle to be `theta=x`. 

If this is true, the parameter shift rule implemented in the `vqregressor` example can't be used as it is. As you can see, in this implementation you divide the shift by `this_feature` if `i % 3 == 0`. This is done because we needed to calculate the derivative of the circuit **only** with respect to the variational parameters $\theta_0$, which is involved into the angle definition like $\theta = \theta_0\cdot x$. 

Here two questions:
1. do you need to calculate the derivative with respect to the first rotational angle? It is not a variational parameter in this case!
2. if yes, you should fix the rule.

As general suggestion, think about the possibility to use [this newer implementation of the PSR](https://github.com/qiboteam/qibo/blob/master/src/qibo/derivative.py) we did.

Reply:
You're right, I do not need to calculate the derivative with respect to the first rotational angle. I've removed that part of the PSR.